File: app/src/main/java/com/cv4j/app/activity/PaintActivity.java
package com.cv4j.app.activity;

import android.os.Bundle;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.support.v7.widget.Toolbar;

import com.cv4j.app.R;
import com.cv4j.app.adapter.PaintAdapter;
import com.cv4j.app.app.BaseActivity;
import com.cv4j.app.fragment.PaintFragment;
import com.safframework.injectview.annotations.InjectExtra;
import com.safframework.injectview.annotations.InjectView;
import com.safframework.injectview.annotations.OnClick;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by Tony Shen on 2017/5/7.
 */

public class PaintActivity extends BaseActivity {

    @InjectView(R.id.tablayout)
    TabLayout mTabLayout;

    @InjectView(R.id.viewpager)
    ViewPager mViewPager;

    @InjectView(R.id.toolbar)
    Toolbar toolbar;

    @InjectExtra(key = "Title")
    String title;

    List<Fragment> mList = new ArrayList();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_oil_paint);

        initData();
    }

    private void initData() {

        toolbar.setTitle("< "+title);

        mList.add(PaintFragment.newInstance(0));
        mList.add(PaintFragment.newInstance(1));
        mList.add(PaintFragment.newInstance(2));
        mViewPager.setAdapter(new PaintAdapter(this, this.getSupportFragmentManager(),mList));
        mTabLayout.setupWithViewPager(mViewPager);
    }

    @OnClick(id= R.id.toolbar)
    void clickToolbar() {

        finish();
    }
}

File: app/src/main/java/com/cv4j/app/activity/PaintActivity.kt
package com.cv4j.app.activity

import android.os.Bundle
import androidx.fragment.app.Fragment
import com.cv4j.app.R
import com.cv4j.app.adapter.PaintAdapter
import com.cv4j.app.app.BaseActivity
import com.cv4j.app.fragment.PaintFragment
import kotlinx.android.synthetic.main.activity_oil_paint.*
import java.util.*

/**
 *
 * @FileName:
 *          com.cv4j.app.activity.PaintActivity
 * @author: Tony Shen
 * @date: 2020-05-05 21:13
 * @version: V1.0 <描述当前版本功能>
 */

class PaintActivity : BaseActivity() {

    var title: String? = null
    private val mList: MutableList<Fragment> = ArrayList<Fragment>()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_oil_paint)

        intent.extras?.let {
            title = it.getString("Title","")
        }?:{
            title = ""
        }()

        toolbar.setOnClickListener {
            finish()
        }

        initData()
    }

    private fun initData() {
        toolbar.title = "< $title"
        mList.add(PaintFragment.newInstance(0))
        mList.add(PaintFragment.newInstance(1))
        mList.add(PaintFragment.newInstance(2))
        viewpager.setAdapter(PaintAdapter(this.getSupportFragmentManager(), mList))
        tablayout.setupWithViewPager(viewpager)
    }
}